xbmc/addons/Skin.cpp

```diff
@@ -369,6 +369,8 @@ const INFO::CSkinVariableString* CSkinInfo::CreateSkinVariable(const std::string
 
 void CSkinInfo::OnPreInstall()
 {
+  if (IsInUse())
+    CApplicationMessenger::GetInstance().SendMsg(TMSG_EXECUTE_BUILT_IN, -1, -1, nullptr, "UnloadSkin");
```

Comment:
The problem is a design issue. One cannot assume that you can just "move" a file that has open handles - we need a mechanism to ensure that everything is properly closed when updating a skin under user's ass :-)